.gitconfig

```diff
@@ -57,6 +57,13 @@
 	# a.k.a. ‘delete merged’
 	dm = "!git branch --merged | grep -v '\\*' | xargs -n 1 git branch -d"
 
+	# Instantly clean a git repo from extraneous files in a non-blocking fashion
+	purge = "!f() { \
+		declare tmp_dir=$(mktemp -d); \
+		git clean -fdxn | sed -n -r 's/^Would remove (.*)/\\1/p' | xargs mv -t \"$tmp_dir\"; \
```

Comment:
>Whenever a command is unwieldy towards xargs, you can always pipe to a while loop and read.

Nice, TIL :+1:.

Utilising the `xargs` replace string flag was actually my first approach, can't remember why I ditched it, might've been due to GNU/BSD incompatibilities. Wanted something platform-agnostic. Actually, now when I take a 2nd look it looks like `-I` does the job too. Updated with something that Should Work Everywhere&trade;.

Btw, do you know if one needs to quote the replace string character to avoid word splitting?